test/unit/test.amo-client.js

```diff
@@ -328,18 +328,57 @@ describe('amoClient.Client', function() {
 
     it('makes requests with an auth token', function(done) {
       var self = this;
+      var request = {url: '/somewhere'};
 
-      this.client.get({url: '/somewhere'}).then(function() {
+      this.client.get(request).then(function() {
         var call = self.client._request.calls[0];
         var headerMatch = call.conf.headers.Authorization.match(/JWT (.*)/);
         var token = headerMatch[1];
         var data = jwt.verify(token, self.client.apiSecret);
         expect(data.iss).to.be.equal(self.client.apiKey);
         expect(data).to.have.keys(['iss', 'iat', 'exp']);
+        expect(call.conf).to.be.deep.equal(
+            self.client.configureRequest(request));
         done();
       }).catch(done);
     });
 
+    it('lets you configure a request directly', function() {
+      var conf = this.client.configureRequest({url: '/path'});
+      expect(conf).to.have.keys(['headers', 'url']);
+      expect(conf.headers).to.have.keys(['Accept', 'Authorization']);
+    });
+
+    it('preserves request headers', function() {
+      var headers = {'X-Custom': 'thing'};
+      var conf = this.client.configureRequest({
+        url: '/path',
+        headers: headers,
+      });
+      expect(conf.headers['X-Custom']).to.be.equal(headers['X-Custom']);
+    });
+
+    it('allows you to override request headers', function() {
+      var headers = {'Accept': 'text/html'};
+      var conf = this.client.configureRequest({
+        url: '/path',
+        headers: headers,
+      });
+      expect(conf.headers['Accept']).to.be.equal(headers['Accept']);
+    });
+
+    it('makes relative URLs absolute', function() {
+      var urlPath = '/somewhere';
+      var conf = this.client.configureRequest({url: urlPath});
+      expect(conf.url).to.be.equal(this.client.apiUrlPrefix + urlPath);
```

Comment:
I moved it to a test instance var which should add some protection against `undefined` https://github.com/kumar303/jpm/blob/fix-download/test/unit/test.amo-client.js#L20